coordinator/app/src/main/kotlin/linea/coordinator/config/v2/toml/decoders/Decoders.kt

```diff
@@ -0,0 +1,83 @@
+package linea.coordinator.config.v2.toml.decoders
+
+import com.sksamuel.hoplite.ConfigFailure
+import com.sksamuel.hoplite.ConfigResult
+import com.sksamuel.hoplite.DecoderContext
+import com.sksamuel.hoplite.Node
+import com.sksamuel.hoplite.StringNode
+import com.sksamuel.hoplite.decoder.Decoder
+import com.sksamuel.hoplite.fp.invalid
+import com.sksamuel.hoplite.fp.valid
+import linea.coordinator.config.v2.toml.SignerConfigToml
+import linea.kotlin.decodeHex
+import kotlin.reflect.KType
+import kotlin.time.Duration
+
+class TomlByteArrayHexDecoder : Decoder<ByteArray> {
+  override fun decode(
+    node: Node,
+    type: KType,
+    context: DecoderContext,
+  ): ConfigResult<ByteArray> {
+    return when (node) {
+      is StringNode -> runCatching {
+        node.value.decodeHex()
+      }.fold(
+        { it.valid() },
+        { ConfigFailure.DecodeError(node, type).invalid() },
+      )
+
+      else -> { ConfigFailure.DecodeError(node, type).invalid() }
+    }
+  }
+
+  override fun supports(type: KType): Boolean {
+    return type.classifier == ByteArray::class
+  }
+}
+
+class TomlKotlinDurationDecoder : Decoder<Duration> {
```

Comment:
I don't see why would you need that. Hoplite supports things like `` out of the box. [Here](https://github.com/Consensys/maru/blob/main/config/src/test/kotlin/maru/config/HopliteFriendlinessTest.kt#L44) is an example in Maru